pkg/serviceaccount/claims.go

```diff
@@ -36,18 +43,19 @@ type privateClaims struct {
 }
 
 type kubernetes struct {
-	Namespace string `json:"namespace,omitempty"`
-	Svcacct   ref    `json:"serviceaccount,omitempty"`
-	Pod       *ref   `json:"pod,omitempty"`
-	Secret    *ref   `json:"secret,omitempty"`
+	Namespace string          `json:"namespace,omitempty"`
+	Svcacct   ref             `json:"serviceaccount,omitempty"`
+	Pod       *ref            `json:"pod,omitempty"`
+	Secret    *ref            `json:"secret,omitempty"`
+	WarnAfter jwt.NumericDate `json:"warnafter,omitempty"`
```

Comment:
I thought omitempty didn't work for types with custom marshalers... maybe that was just struct types, and this is a type alias of an int

might be clearer to make this `WarnAfter *jwt.NumericDate` and only set it when non-zero... that makes it clearer it is optional and that omitempty will work as expected

Reply:
NumericDate is just an alias of int64
I kind of copied how we define such field from "gopkg.in/square/go-jose.v2/jwt" package, it defines exp, iat like this.